### ISSUE TITLE
Setting `csp.upgradeInsecureRequests` on breaks on startup

### ISSUE THREAD

**Issue by mdegat01** [hedgedoc/hedgedoc#1221]
**Description**

I was not specifying anything for `csp` in my hedgedoc configuration which means I was using the default value for this setting as specified in the [docs](https://docs.hedgedoc.org/configuration/#web-security-aspects):

```json
{"enable": true, "directives": {"scriptSrc": "trustworthy-scripts.example.com"}, "upgradeInsecureRequests": "auto", "addDefaults": true}
```

I also have `useSSL` set to `true` so that means `auto` kicks in and becomes `true` based on the code [here](https://github.com/hedgedoc/hedgedoc/blob/f48e36d2052bcae56081bee3c281bfc69fe7f70d/lib/csp.js#L86). When I updated to 1.8.0 it broke on startup with this error:

```
2021-05-03T22:15:26.943Z error: 	uncaughtException: directiveValue is not iterable
TypeError: directiveValue is not iterable
    at normalizeDirectives (/opt/hedgedoc/node_modules/helmet/dist/middlewares/content-security-policy/index.js:64:25)
    at Function.contentSecurityPolicy (/opt/hedgedoc/node_modules/helmet/dist/middlewares/content-security-policy/index.js:117:31)
    at Object.<anonymous> (/opt/hedgedoc/app.js:123:18)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
```

I began playing around with the config and managed to isolate the error to the `upgradeInsecureRequests` option. If I set this to `"auto"` or `true` then it breaks on startup with this error. If I set this to `false` then it works fine.

Some further poking around makes me think that particular directive is being set wrong. The code which seems to set that directive for helmet seems to set it to the  literal value `true` when `upgradeInsecureRequests` is turned on. But in Helmet's [documentation](https://helmetjs.github.io/) this is their example which uses `upgradeInsecureRequests`:

```js
// Sets "Content-Security-Policy: default-src 'self';script-src 'self' example.com;object-src 'none';upgrade-insecure-requests"
app.use(
  helmet.contentSecurityPolicy({
    useDefaults: false,
    directives: {
      defaultSrc: ["'self'"],
      scriptSrc: ["'self'", "example.com"],
      objectSrc: ["'none'"],
      upgradeInsecureRequests: [],
    },
  })
);
```

Seems like the value should be set to `[]` rather then `true`.
<s>EDIT: This isn't it. I tried this change and it still broke. I can only confirm that its related to that config, can't figure out why. I'll have to leave this one to the experts.</s>
EDIT: Seems this was it. I'm sure I just screwed up my test, glad others got to the bottom of it.

**To Reproduce**

Shouldn't be an error

**Logs**

```
2021-05-03T22:15:26.943Z error: 	uncaughtException: directiveValue is not iterable
TypeError: directiveValue is not iterable
    at normalizeDirectives (/opt/hedgedoc/node_modules/helmet/dist/middlewares/content-security-policy/index.js:64:25)
    at Function.contentSecurityPolicy (/opt/hedgedoc/node_modules/helmet/dist/middlewares/content-security-policy/index.js:117:31)
    at Object.<anonymous> (/opt/hedgedoc/app.js:123:18)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
```

**Config**

```json
{
  "production": {
    "csp": {
      "enable": true,
      "upgradeInsecureRequests": false,
      "addDefaults": true
    }
}
```

**Your Setup (please complete the following information):**
 - Host OS: Alpine 3.13.4
 - NodeJS version: v14.16.1
 - HedgeDoc version (Click "Version info" in the footer)
     - server: 1.8.0
     - client: 1.8.0

**Additional context**

**Comment by mdegat01** [comment]
Ok well I was wrong about the `[]` vs. `true` value being the problem. I just tested that locally and I still get this error. It's still related to this config I just don't know why. Was hoping I could submit a quick PR to help out but my expertise is pretty limited here.

**Comment by davidmehren** [comment]
Thanks for reporting this!

In my testing, using `[]` instead of `true` fixes the issue, as helmet tries to iterate over the values of the directives at https://github.com/helmetjs/helmet/blob/main/middlewares/content-security-policy/index.ts#L140, which won't work with `true`.

**Comment by davidmehren** [comment]
@mdegat01 Could you have a look at https://github.com/hedgedoc/hedgedoc/pull/1222 and see if that fixes the issue for you?

**Comment by GuillaumeFillol** [comment]
Same here, to make it works I updated config.json from :
```
   csp": {
         "enable": true,
         "directives": {
          },
         "upgradeInsecureRequests": true,
...
    },
```
to :
```
   csp": {
         "enable": true,
         "directives": {
             "upgradeInsecureRequests": true
          },
...
    },
```

**Comment by mdegat01** [comment]
@davidmehren yep that did the trick. That was exactly what I tried to test as the fix, I must've screwed something up lol. Thanks for the quick fix!

**Comment by davidmehren** [comment]
A fix for this will be released with 1.8.1.
